lib/myaccount/request.ts

```diff
@@ -26,23 +27,44 @@ export async function sendRequest<
   const {
     accessToken: accessTokenObj
   } = oktaAuth.tokenManager.getTokensSync();
-  
-  const accessToken = options.accessToken || accessTokenObj?.accessToken;
+
+  const atToken = options.accessToken || accessTokenObj;
   const issuer = oktaAuth.getIssuerOrigin();
   const { url, method, payload } = options;
   const requestUrl = url.startsWith(issuer!) ? url : `${issuer}${url}`;
 
-  if (!accessToken) {
+  if (!atToken) {
     throw new AuthSdkError('AccessToken is required to request MyAccount API endpoints.');
   }
-  
-  const res = await httpRequest(oktaAuth, {
+
+  let accessToken: string | AccessToken = atToken;
+
+  const httpOptions: HttpRequestOptions = {
     headers: { 'Accept': '*/*;okta-version=1.0.0' },
-    accessToken,
     url: requestUrl,
     method,
     ...(payload && { args: payload })
-  });
+  };
+
+  if (oktaAuth.options.dpop) {
+    if (typeof accessToken === 'string') {
+      throw new AuthSdkError('AccessToken object must be provided when using dpop');
+    }
+
+    const { Authorization, Dpop } = await oktaAuth.getDPoPAuthorizationHeaders({
+      method,
+      url: requestUrl,
+      accessToken
+    });
+    httpOptions.headers!.Authorization = Authorization;
+    httpOptions.headers!.Dpop = Dpop;
```

Comment:
nit: I know we can safely use the `!` here to assert that `headers` is defined, but I think using `?.` doesn't hurt us and prevents accidental footguns in the future if more code is added in between here and the definition of `httpOptions` that mutates it

Reply:
TS didn't allow `?`, I tried that originally

Reply:
Oh right, this is an assignment statement so you can't use `?.` on the left side

Another option would be to guard these with `if (httpOptions.headers)`